addons/test_website/tests/test_controller_args.py

```diff
@@ -29,3 +31,17 @@ def test_crawl_args(self):
         req = self.url_open('/ignore_args/kw?a=valueA&b=valueB')
         self.assertEqual(req.status_code, 200)
         self.assertEqual(req.json(), {'a': 'valueA', 'kw': {'b': 'valueB'}})
+
+    def test_route_type(self):
+        response = self.url_open('/test_route_json_http?blabla=qsd')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.text, "{'blabla': 'qsd'}")
+
+        response = self.url_open('/test_route_json_http', data=b"blabla=qsd", headers={'Content-type': 'application/x-www-form-urlencoded'})
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.text, "{'blabla': 'qsd'}")
+
+        response = self.url_open('/test_route_json_http', data=b'{"a": "blabla"}', headers={'Content-type': 'application/json'})
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue("{'a': 'blabla'}" in response.text)
```

Comment:
I guess this comment was lost in a push-force, but I was asking for some improvements of the test logic here:
- test the actual JSON response with `response.json['result']`
- perhaps have the route return structured data in the JSON case, to show a more meaningful use case (here it is almost identical to the HTTP case, which I found confusing while reading)